docs/tasks/extend-kubectl/kubectl-plugins.md

```diff
@@ -0,0 +1,137 @@
+specification. Specifically, the loader locates the directories specified by the `XDG_DATA_DIRS` environment variable,
+and then searches `kubectl/plugins` directory inside of those.
+If `XDG_DATA_DIRS` is not specified, it defaults to `/usr/local/share:/usr/share`.
+
+Second, the `plugins` directory under the user's kubeconfig dir. In most cases, this is `~/.kube/plugins`.
+
+```shell
+# Loads plugins from both /path/to/dir1 and /path/to/dir2
+KUBECTL_PLUGINS_PATH=/path/to/dir1:/path/to/dir2 kubectl plugin -h
+```
+
+## Writing kubectl plugins
+
+You can write a plugin in any programming language or script that allows you to write command-line commands.
+A plugin does not necessarily need to have a binary component. It could rely entirely on operating system utilities
+like `echo`, `sed`, or `grep`. Or it could rely on the `kubectl` binary.
+
+The only strong requirement for a `kubectl` plugin is the `plugin.yaml` descriptor file. This file is responsible for declaring at least the minimum attributes required to register a plugin and must be located under one of the locations specified in the [Search order](#search-order) section. 
+
+### The plugin.yaml descriptor
+
+The descriptor file supports the following attributes:
+
+```
+name: "targaryen"                 # REQUIRED: the plugin command name, to be invoked under 'kubectl'
+shortDesc: "Dragonized plugin"    # REQUIRED: the command short description, for help
+longDesc: ""                      # the command long description, for help
+example: ""                       # command example(s), for help
+command: "./dracarys"             # REQUIRED: the command, binary, or script to invoke when running the plugin
+flags:                            # flags supported by the plugin
+  - name: "heat"                  # REQUIRED for each flag: flag name
+    shorthand: "h"                # short version of the flag name
+    desc: "Fire heat"             # REQUIRED for each flag: flag description
+    defValue: "extreme"           # default value of the flag
+tree:                             # allows the declaration of subcommands
+  - ...                           # subcommands support the same set of attributes
+```
+
+The preceding descriptor declares the `kubectl plugin targaryen` plugin, which has one flag named `-h | --heat`.
+When the plugin is invoked, it calls the `dracarys` binary or script, which is located in the same directory as the descriptor file. The [Accessing runtime attributes](#accessing-runtime-attributes) section describes how the `dracarys` command accesses the flag value and other runtime context.
+
+### Recommended directory structure
+
+It is recommended that each plugin has its own subdirectory in the filesystem, preferably with the same name as the plugin command. The directory must contain the `plugin.yaml` descriptor and any binary, script, asset, or other dependency it might require.
+
+For example, the directory structure for the `targaryen` could look like this:
```

Comment:
For example, the directory structure for the `targaryen` could look like this:
->
For example, the directory structure for plugin `targaryen` could look like this:
or
For example, the directory structure for plugin `targaryen`:
These would look fine. =) I think.